packages/mwp-router/src/util/resolve.js

```diff
@@ -138,3 +135,23 @@ export const getRouteResolver = (
 	const path = location.pathname.replace(basename, '');
 	return _resolveRouteMatches(routes, path);
 };
+
+export const resolveRoute = (route: PlatformRoute): Promise<PlatformRoute> => {
+	return _resolveIndexRoute(route)
+		.then(_resolveNestedRoutes)
+		.then(resolveComponent)
+		.then(({ routes, getComponent, ...noRoutesRoute }: PlatformRoute) =>
+			resolveAllRoutes(routes || []).then((routes: Array<PlatformRoute>) =>
+				Object.freeze({
+					...noRoutesRoute,
+					routes,
+				})
+			)
+		);
+};
+
+export const resolveAllRoutes = (
```

Comment:
TODO: document these functions. They are server-side only because they "over-resolve" - i.e. they resolve all the Promises in the routing tree, not just the ones that correspond to a supplied route like `getRouteResolver` does - going to be reworking that in a new PR